src/store/audio.ts

```diff
@@ -2060,17 +2063,62 @@ export const audioCommandStore = transformCommandStore(
             );
 
             // 読みの内容が変わっていなければテキストだけ変更
-            const isChangedMora = isAccentPhrasesTextDifferent(
+            const isTextDifferent = isAccentPhrasesTextDifferent(
               query.accentPhrases,
               accentPhrases
             );
+            let newAccentPhrases: AccentPhrase[] = [];
+            if (isTextDifferent) {
+              if (state.experimentalSetting.shouldKeepTuningOnTextChange) {
+                const diff = diffArrays(
+                  query.accentPhrases.map(joinTextsInAccentPhrases),
+                  accentPhrases.map(joinTextsInAccentPhrases)
+                );
+                const flatDiff = diff.flatMap((d) =>
+                  d.value
+                    .filter((v) => v !== "")
+                    .map((v) => ({ ...d, value: v }))
+                );
+                const indexedDiff = flatDiff.map((d, i) => ({
+                  ...d,
+                  index: i,
+                }));
+                const indexToOldAccentPhrase = indexedDiff
+                  .filter((d) => !d.added)
+                  .reduce(
+                    (acc, d, i) => ({
+                      ...acc,
+                      [d.index]: toRaw(query.accentPhrases[i]),
+                    }),
+                    {} as { [index: number]: AccentPhrase }
+                  );
+                newAccentPhrases = indexedDiff
+                  .filter((d) => !d.removed)
+                  .map((d, i) => {
+                    const original = accentPhrases[i];
+                    const ap = structuredClone(
+                      indexToOldAccentPhrase[d.index] ?? original
+                    );
+                    if (accentPhrases[i].pauseMora !== undefined) {
+                      ap.pauseMora = original.pauseMora;
+                    } else {
+                      delete ap.pauseMora;
+                    }
+                    ap.isInterrogative = original.isInterrogative;
+
+                    return ap;
```

Comment:
originalだと前のか後のかどっちの話か分からないので、oldに合わせてnewAccentPhraseとかにするのどうでしょう？
あるいはそのまま`accentPhrases[i]`使っちゃう手もありそう。
```suggestion
                    const ap = structuredClone(
                      indexToOldAccentPhrase[d.index] ?? accentPhrases[i]
                    );
                    if (accentPhrases[i].pauseMora !== undefined) {
                      ap.pauseMora = accentPhrases[i].pauseMora;
                    } else {
                      delete ap.pauseMora;
                    }
                    ap.isInterrogative = original.isInterrogative;

                    return ap;
```